t/test.pl

```diff
@@ -2022,6 +2009,9 @@ ($;$)
                 kill($sig, $pid_to_kill);
         });
 
+        $cancel_string =~ s/HERE/$watchdog_thread->kill('KILL')->detach()/;
```

Comment:
`$watchdog_thread` would be replaced at this point, producing code like `threads=SCALAR(0x55e3bd6a7ab0)->kill('KILL')->detach()` which will just fail to compile.

Reply:
This was wishful thinking on my part